src/pause/pause_spirits.c

```diff
@@ -0,0 +1,371 @@
+                                          + (panel->numCols * panel->row)
+                                          + (panel->col)]) {
+                                              break;
+                                          }
+        }
+    }
+
+    if (gPauseHeldButtons & BUTTON_STICK_RIGHT) {
+        while (TRUE) {
+            panel->col++;
+            if (panel->col >= panel->numCols) {
+                panel->col = panel->numCols - 1;
+                break;
+            }
+
+            if (panel->selected != panel->gridData[
+                                            (panel->page * panel->numCols * panel->numRows)
+                                          + (panel->numCols * panel->row)
+                                          + (panel->col)]) {
+                                              break;
+                                          }
+        }
+    }
+
+    if (gPauseHeldButtons & BUTTON_STICK_UP) {
+        if (panel->selected == 2) {
+            panel->col = 3;
+            panel->row = 0;
+        } else if (panel->selected == 5) {
+            panel->col = 1;
+            panel->row = 0;
+        } else {
+            while (TRUE) {
+                panel->row--;
+                if (panel->row < 0) {
+                    panel->row = 0;
+                    break;
+                }
+                if (panel->selected != panel->gridData[
+                                                (panel->page * panel->numCols * panel->numRows)
+                                              + (panel->numCols * panel->row)
+                                              + (panel->col)]) {
+                                                  break;
+                                              }
+            }
+        }
+    }
+
+    if (gPauseHeldButtons & BUTTON_STICK_DOWN) {
+        if (panel->selected == 2) {
+            panel->col = 3;
+            panel->row = 1;
+        } else if (panel->selected == 5) {
+            panel->col = 1;
+            panel->row = 1;
+        } else {
+            while (TRUE) {
+                panel->row++;
+                if (panel->row >= panel->numRows) {
+                    panel->row = panel->numRows - 1;
+                    break;
+                }
+                if (panel->selected != panel->gridData[
+                                                (panel->page * panel->numCols * panel->numRows)
+                                              + (panel->numCols * panel->row)
+                                              + (panel->col)]) {
+                                                  break;
+                                              }
+            }
+        }
+    }
+
+    panel->selected = panel->gridData[
+                              (panel->page * panel->numCols * panel->numRows)
+                            + (panel->numCols * panel->row)
+                            + (panel->col)];
```

Comment:
This appears a lot, you could maybe make a macro for it. @ethteck - thoughts? It would clean up those ifs above too.

Reply:
yeah that sounds good to me